### PR TITLE
Send Share Action Facebook Posts to Rogue with only Action ID

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request updates the `ShareAction` component to send Facebook social shares to Rogue for authenticated users when there's an Action ID present, even if there's no Campaign ID.

It also adds a couple of Cypress tests (the first few commits are chores around setting us up for these new tests) per this new functionality and adds some documentation.

### How should this be reviewed?
I think commit-by-commit would be easiest. Are the changes for the Cypress tests looking good? is the new Share Action logic sound?

### Any background context you want to provide?
We added the Action ID in https://github.com/DoSomething/phoenix-next/pull/1256 when the Share Action only used the Campaign ID (and subsequently the `action` as well) to tie social share posts with an Action in Rogue.

This does mean that only Share Actions within a _Campaign context_ (i.e. campaign pages or campaign block pages like `/us/campaigns/:slug/blocks/:id`) would track to Rogue for authed users.

This allows us to embed the Share Action on an independent blocks page (`/us/blocks/:id`) or a Story page, and still post to Rogue for authenticated users.

### Relevant tickets

References [Pivotal #175442499](https://www.pivotaltracker.com/story/show/175442499).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [x] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: cypress/fixtures/constants.js [hunk 1]
```diff
@@ -9,3 +9,8 @@ export const defaultReferralCampaignId = '9001';
 export const campaignPath = '/us/campaigns/';
 
 export const userId = '5575e568a59dbf3b7a8b4572';
+
+// Rouge API paths.
```
Comment (aaronschachter) [inline cypress/fixtures/constants.js:13]: 🎲 
```suggestion
// Rogue API paths.
```

File: resources/assets/components/actions/ShareAction/ShareAction.js [hunk 2]
```diff
@@ -102,8 +110,12 @@ class ShareAction extends PostForm {
...
-        if (this.props.isAuthenticated && this.props.campaignId) {
+        // Send share post to Rogue for authenticated users,
+        // when we have an action ID or campaign ID to identify this post.
+        if (
+          this.props.isAuthenticated &&
+          (this.props.campaignId || this.props.actionId)
+        ) {
           const puckId = `phoenix_${userId}_${Date.now()}`;
 
           trackingData = { ...trackingData, puck_id: puckId };
```
Comment (aaronschachter) [inline resources/assets/components/actions/ShareAction/ShareAction.js:121]: Puck? RIP

File: cypress/fixtures/constants.js (hunk 1 above)
Comment (aaronschachter) [reply cypress/fixtures/constants.js:13]: Aren't these Phoenix paths, not Rogue paths?

File: docs/development/content-types/share-action.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Share Action
+
+## Overview
+
+Displays a card with text, a link embed, and a button to share the link on social media (Twitter or Facebook).
+
+![Example Social Drive Action](../../.gitbook/assets/share-action.png)
```
Comment (aaronschachter) [inline docs/development/content-types/share-action.md:7]:
```suggestion
![Example Share Action](../../.gitbook/assets/share-action.png)
```

**Review by aaronschachter — APPROVED** [review]
Looks good! Great job commenting/documenting.

File: cypress/fixtures/constants.js (hunk 1 above)
Comment (mendelB) [reply cypress/fixtures/constants.js:13]: That's true! I'll update

File: resources/assets/components/actions/ShareAction/ShareAction.js (hunk 2 above)
Comment (mendelB) [reply resources/assets/components/actions/ShareAction/ShareAction.js:121]: lolol! I'll add a comment to look into if we can deprecate this. (Presumably, we can... We added this for https://www.pivotaltracker.com/story/show/160100106)

**Pull request merged**